pkg/utils/node/finalizer.go

```diff
@@ -0,0 +1,17 @@
+package node
+
+import (
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func RemoveFinalizer(node *v1.Node, index int) {
```

Comment:
Consider turning the slice into a map[string]bool, deleting the key you don't want, and then reconverting. This is the go idiomatic way to use a set.